src/components/password.components.js

```diff
@@ -0,0 +1,98 @@
+import $ from 'jquery';
+
+export default class PasswordInput {
+    constructor(queryString, valid = false, required = true, disabled = false) {
+        this.div = $(queryString);
+        this.input = $(`${queryString} input[type="password"]`);
+        this.icon = $(`${queryString} input[type="password"]+i`);
+
+        this.valid = valid;
+        this.disabled = disabled;
+        this.required = required;
+
+        this.hasQuery(this.div, this.input, this.icon);
+        this.initFor();
+    }
+
+    hasQuery(div, input, icon) {
+        if (div.length === 0) {
+            throw new Error('Failed to query div');
+        }
+
+        if (input.length === 0) {
+            throw new Error('Failed to query input password');
+        }
+
+        if (icon.length === 0) {
+            throw new Error('Failed to query icon');
+        }
+    }
+
+    initFor() {
+        this.icon.on('click', () => {
+            if (this.input.attr('type') === 'password') {
+                this.input.attr('type', 'text').trigger('focus');
+                this.icon.removeClass('fa-eye-slash').addClass('fa-eye');
+            } else {
+                this.input.attr('type', 'password').trigger('focus');
+                this.icon.removeClass('fa-eye').addClass('fa-eye-slash');
+            }
+        });
+    }
+
+    getValue() {
+        return this.input.val();
+    }
+
+    setValue(string) {
+        this.input.val(string);
+    }
+
+    isValid() {
+        // Validate regex: 1 capital letter, 1 number, 1 special character, and size 8;
+        const validations = /^(?=.*[A-Z])(?=.*\d)(?=.*[!"#$%&()*,.:<>?@^{|}])[\d!"#$%&()*,.:<>?@A-Z^a-z{|}]{8,}$/;
+
+        if (validations.test(this.input.val())) {
+            this.valid = true;
+            return this.valid;
+        }
+
+        return this.valid;
+    }
+
+    getDisabled() {
+        return this.input.prop('disabled');
+    }
+
+    setDisabled(boolean) {
```

Comment:
This function when passing a value true changes the style:
![image](https://user-images.githubusercontent.com/92385147/162186991-0e702b1a-d7d3-44c8-a954-1422ae1c11eb.png)
I can also click on the icon to show or hide a password:
![image](https://user-images.githubusercontent.com/92385147/162187161-69769cc8-238c-46d7-9e37-35da99883d32.png)